ui/apps/platform/src/Containers/NetworkGraph/TopologyComponent.tsx

```diff
@@ -34,6 +34,60 @@ import {
 import { EdgeState } from './components/EdgeStateSelect';
 import { deploymentTabs } from './utils/deploymentUtils';
 import { NetworkScopeHierarchy } from './types/networkScopeHierarchy';
+import { EntityScope } from './simulation/NetworkPoliciesGenerationScope';
+
+/**
+ * Given the array of nodeModels returned by the network graph API and the user's
+ * current scope hierarchy, return the scope of entities that should be included
+ * in the network policy simulation.
+ */
+function getInScopeEntities(
+    nodeModels: CustomNodeModel[],
+    scopeHierarchy: NetworkScopeHierarchy
+): EntityScope {
+    let granularity: EntityScope['granularity'] = 'CLUSTER';
+
+    if (scopeHierarchy.namespaces.length === 0 && scopeHierarchy.deployments.length === 0) {
+        // Only a cluster is selected
+        granularity = 'CLUSTER';
+    } else if (scopeHierarchy.namespaces.length > 0 && scopeHierarchy.deployments.length === 0) {
+        // A cluster and at least one namespace is selected
+        granularity = 'NAMESPACE';
+    } else {
+        // A cluster, at least one namespace, and at least one deployment is selected
+        granularity = 'DEPLOYMENT';
+    }
+
+    // A cluster, at least one namespace, and at least one deployment is selected. We
+    // include all deployments from the model that match a selected namespace and
+    // selected deployment name from the scope hierarchy.
+    const namespaceNameSet = new Set(scopeHierarchy.namespaces);
+    const deploymentNameSet = new Set(scopeHierarchy.deployments);
+
+    const deploymentScope: EntityScope = {
+        granularity,
+        cluster: scopeHierarchy.cluster.name,
+        namespaces: scopeHierarchy.namespaces,
```

Comment:
why not use `namespaceNameSet` here instead?

Reply:
No particular reason, `scopeHierarchy.namespaces` and `namespaceNameSet` should always contain the same items. The `Set` creation was just for efficiency in checking if a namespace is in the selected list of namespaces (probably early optimization though).